api/python/notebooks/analysis_demo/comp_bio_query_data_and_metadata.ipynb

```diff
@@ -0,0 +1,1056 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Querying and fetching data from the Cell Census\n",
+    "\n",
+    "The Cell Census is a versioned container for the single-cell data hosted at [CELLxGENE Discover](https://cellxgene.cziscience.com/). The Cell Census utilizes [SOMA](https://github.com/single-cell-data/SOMA/blob/main/abstract_specification.md) powered by [TileDB](https://tiledb.com/products/tiledb-embedded) for storing, accessing, and efficiently filtering data.\n",
+    "\n",
+    "This notebook showcases the easiest ways to query the expression data and cell/gene metadata from the Cell Census.\n",
+    "\n",
+    "\n",
+    "# Contents\n",
+    "- Opening the census\n",
+    "- Querying cell metadata (obs)\n",
+    "- Querying gene metadata (var)\n",
+    "- Querying expression data\n",
+    "\n",
+    "## Opening the census\n",
+    "\n",
+    "The `cell_census` python package contains a convenient API to open the latest version of the Cell Census."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cell_census\n",
+    "\n",
+    "census = cell_census.open_soma()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can learn more about the `cell_census` methods by accessing their corresponding documentation via `help()`. For example `help(cell_census.open_soma)`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Querying cell metadata (obs)\n",
+    "\n",
+    "The human cell metadata of the Cell Census is located at `census[\"census_data\"][\"homo_sapiens\"].obs`. This is a `SOMADataFrame` and as such it can be materialized as a `pandas.DataFrame` via the method `read_as_pandas_all()`. \n",
+    "\n",
+    "The mouse cell metadata is at `census[\"census_data\"][\"mus_musculus\"].obs`.\n",
+    "\n",
+    "For slicing the cell metadata there are two relevant arguments that can be passed through `read_as_pandas_all()`:\n",
+    "\n",
+    "- `column_names` — list of strings indicating what metadata columns to fetch. \n",
+    "- `value_filter` — python expression with selection conditions to fetch rows, it is similar to [`pandas.DataFrame.query()`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.query.html).\n",
```

Comment:
This needs a link to a description of the actual syntax supported by the `value_filter`.  At the moment, the only place I can find it today is in the docstring for `tiledb.QueryCondition`.  I have filed an issue about this in single-cell-data/TileDB-SOMA#612.

In the short term, it might be worth adding a summary of that docstring to this notebook, as it is fairly brief & simple, i.e.,
* expressions are one or more comparisons
* comparisons are one of `column op value` or `column op column`
* expressions can combine comparisons using `and`, `or`, `&` or `|`
* op is one of `< | > | <= | >= | == | !=` or `in`
etc.

`help(tiledb.QueryCondition)` has the details currently available if you want to do a brief placeholder.

Reply:
also found: https://tiledb-inc-tiledb.readthedocs-hosted.com/projects/tiledb-py/en/stable/python-api.html#query-condition

Reply:
Thanks I added this info